mesonbuild/ast/introspection.py

```diff
@@ -137,17 +142,20 @@ def do_subproject(self, dirname: str) -> None:
             return
 
     def func_add_languages(self, node: BaseNode, args: T.List[TYPE_nvar], kwargs: T.Dict[str, TYPE_nvar]) -> None:
-        args = self.flatten_args(args)
         for for_machine in [MachineChoice.BUILD, MachineChoice.HOST]:
```

Comment:
And we can do something based off of `kwargs` for this list too? Granted that a pre-existing bug we could also punt for a future PR. I know you've done tons of these revisions already :).

Reply:
I think I omitted that logic, because as far as I can see, the native kwarg is ignored in the introspector for all other meson functions which support it (so presumably isn't signficant to introspection)

Reply:
Adding `native` support here would be nice, if it would change the introspection output (what meson options printed with `meson configure` without a build dir). The other functions don't use `native` since it wouldn't make any difference to the introspection output (as far as I know).

Reply:
It seems a bit odd that running the introspector on an unconfigured meson.build will detect compilers, but I guess that makes sense.